ggshield/cmd/utils/common_options.py

```diff
@@ -207,6 +207,23 @@ def _set_json_output_format(
 )
 
 
+def _set_output_format(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[str]:
+    if value:
+        ctx_obj = ContextObj.get(ctx)
+        ctx_obj.output_format = OutputFormat(value)
+    return value
+
+
+format_option = click.option(
+    "--format",
+    type=click.Choice([x.value for x in OutputFormat]),
+    help="Output format.",
+    callback=_set_output_format,
+)
```

Comment:
I really like the addition of the `--format` option but we should note that it creates different behavior depending on the flags order:
* `--format sarif --json` => json output 
* `--json --format sarif` => sarif output
* `--format json --format sarif` => sarif output
I am concerned with confusion for users (some may think we could output to several format at the same time), do we want to clarify that in our help message ? Or even fail when  we are given several options ?

Reply:
Also, maybe you could add a changelog for the addition of this option ?

Reply:
> I really like the addition of the --format option but we should note that it creates different behavior depending on the flags order

Good point! I don't want to make it too complex by failing if both are set, but I am going to improve the --help output to mention that the last one set wins.

> Also, maybe you could add a changelog for the addition of this option ?

Oh yes, definitely!